packages/common/src/TransactionUtils.ts

```diff
@@ -19,6 +19,10 @@ export interface AugmentedSendOptions {
   maxPriorityFeePerGas?: number | string;
 }
 
+interface AugmentedWeb3 extends Web3 {
```

Comment:
to accommodate the nonce management, we append a `nonces` object to the web3 interface. This acts as a global nonce counter.